pkg/registry/core/service/portallocator/operation.go

```diff
@@ -95,6 +100,8 @@ func (op *PortAllocationOperation) Commit() []error {
 
 // Allocates a port, and record it for future rollback
 func (op *PortAllocationOperation) Allocate(port int) error {
```

Comment:
Is `op` called in a threaded context?  If so, this change would be correct.  Allocate *should* be thread safe and only allocate a port once, but double checking that would be good.

Reply:
I do not think so. It is called in https://github.com/kubernetes/kubernetes/blob/master/pkg/registry/core/service/rest.go